pkg/authorization/workspace_content_authorizer.go

```diff
@@ -114,6 +114,19 @@ func (a *workspaceContentAuthorizer) Authorize(ctx context.Context, attr authori
 	isServiceAccountFromRootCluster := subjectClusters[tenancyv1alpha1.RootCluster]
 	isServiceAccountFromCluster := subjectClusters[cluster.Name]
 
+	if IsDeepSubjectAccessReviewFrom(ctx, attr) {
+		attr := deepCopyAttributes(attr)
+		// this is a deep SAR request, we have to skip the checks here and delegate to the subsequent authorizer.
+		if isAuthenticated && !isUser && !isServiceAccountFromCluster {
+			// anonymize service accounts from other workspaces
```

Comment:
Why is this important?

Reply:
think `default` service account arrives here, without anonymization he might receive unwanted permissions.

Reply:
Can we add a comment explaining the rationale?

Reply:
oh definitely, this makes sense, yes, fixed